IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ConditionalWait.cs

```diff
@@ -0,0 +1,113 @@
+﻿using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IntelliTect.TestTools.Selenate
+{
+    public class ConditionalWait
```

Comment:
One final naming comment. I would not include "ForSeconds" as part of the method name. That information is already captured with the `seconds` parameter. In addition, because it is an optional parameter, it looks a little weird to see an invocation that has `.WaitForSeconds(() => ...);` but never specifies any seconds. If you want it to be clear to the reader what the second parameter is, consider using named parameters or changing the type to `TimeSpan`.